### PR TITLE
TW-1091: Migration from FlatList to FlashList

### PR DESCRIPTION
**Pull request by lendihop**
https://madfish.atlassian.net/jira/software/projects/TW/boards/4?selectedIssue=TW-1091

### CODE REVIEW

File: src/components/activity-groups-list/activity-groups-list.tsx [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
 import { useActivityGroupsListStyles } from './activity-groups-list.styles';
 
+type ListItem = string | ActivityGroup;
+
+const keyExtractor = (item: ListItem) => (typeof item === 'string' ? item : item[0].hash);
+const getItemType = (item: ListItem) => (typeof item === 'string' ? 'sectionHeader' : 'row');
+
+const ListEmptyComponent = <DataPlaceholder text="No Activity records were found" />;
+
+const AVERAGE_ITEM_HEIGHT = 150;
```
Comment (keshan3262) [inline src/components/activity-groups-list/activity-groups-list.tsx:24]: Did you mean `Math.floor(formatSize(150))`?

File: src/components/activity-groups-list/activity-groups-list.tsx [hunk 2]
```diff
@@ -30,22 +41,23 @@ export const ActivityGroupsList: FC<Props> = ({
...
 
     for (const activityGroup of activityGroups) {
       const firstActivity = activityGroup[0] ?? emptyActivity;
       const date = new Date(firstActivity.timestamp);
+      const lastActivityGroup = result[result.length - 1];
 
-      if (isTheSameDay(date, prevActivityDate)) {
-        result[result.length - 1].data.push(activityGroup);
+      if (isTheSameDay(date, prevActivityDate) && Array.isArray(lastActivityGroup)) {
+        lastActivityGroup.concat(activityGroup);
```
Comment (keshan3262) [inline src/components/activity-groups-list/activity-groups-list.tsx:53]: Sorry, `concat` function doesn't mutate an array instance
```suggestion
        result[result.length - 1] = lastActivityGroup.concat(activityGroup);
```

File: src/components/activity-groups-list/activity-groups-list.tsx [hunk 3]
```diff
@@ -54,49 +66,64 @@ export const ActivityGroupsList: FC<Props> = ({
...
+  const stickyHeaderIndices = useMemo(
+    () =>
+      sections
+        .map((item, index) => {
+          if (typeof item === 'string') {
+            return index;
+          } else {
+            return null;
+          }
+        })
```
Comment (keshan3262) [inline src/components/activity-groups-list/activity-groups-list.tsx:105]:
```suggestion
        .map((item, index) => typeof item === 'string' ? index : null)
```

File: src/screens/d-apps/others/others.tsx [hunk 4]
```diff
@@ -15,31 +15,38 @@ import { useOpenUrlInAppBrowser } from 'src/utils/linking';
...
 
+  const styles = useOthersDAppStyles();
   const openUrl = useOpenUrlInAppBrowser();
+
   const onPress = useCallback(() => openUrl(dappUrl), [openUrl, dappUrl]);
 
   return (
     <TouchableWithAnalytics
-      style={styles.container}
+      style={{
```
Comment (keshan3262) [inline src/screens/d-apps/others/others.tsx:33]: Please add this style to `useOthersDAppStyles`

File: src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-rewards-list.tsx [hunk 5]
```diff
@@ -22,15 +24,17 @@ const allRewardsPerEventKeys: (keyof RewardsPerEventHistoryItem)[] = [
   'rewardPerFutureEndorsement'
 ];
 
+const AVERAGE_ITEM_HEIGHT = 280;
```
Comment (keshan3262) [inline src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-rewards-list.tsx:27]: Did you mean `Math.floor(formatSize(280))`?

File: src/screens/notifications/notifications.tsx [hunk 6]
```diff
@@ -1,29 +1,27 @@
...
+import { ScreensEnum } from 'src/navigator/enums/screens.enum';
+import { viewAllNotificationsAction } from 'src/store/notifications/notifications-actions';
+import { useNotificationsSelector } from 'src/store/notifications/notifications-selectors';
+import { usePageAnalytic } from 'src/utils/analytics/use-analytics.hook';
+
 import { NotificationPreviewItem } from './notification-preview-item/notification-preview-item';
 import { NotificationsStyles } from './notifications.styles';
 
 const VIEW_ALL_NOTIFICATIONS_TIMEOUT = 5 * 1000;
+const AVERAGE_NOTIFICATION_ITEM_HEIGHT = 127;
```
Comment (keshan3262) [inline src/screens/notifications/notifications.tsx:16]: Did you mean `Math.floor(formatSize(127))`?

File: src/components/activity-groups-list/activity-groups-list.tsx (hunk 1 above)
Comment (lendihop) [reply src/components/activity-groups-list/activity-groups-list.tsx:24]: docs for `estimatedItemSize` prop: 
If the items have a lot of different sizes, choose a few that are pretty different, use element inspector, read their sizes, and calculate an average. This average will work great, don't worry about solving for different devices sizes. There's enough tolerance internally.

so i think there is no need of `formatSize` and `Math.floor` in this case

File: src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-rewards-list.tsx (hunk 5 above)
Comment (lendihop) [reply src/screens/delegation-screen/selected-baker-screen/baker-rewards-list/baker-rewards-list.tsx:27]: look at the 1 comment thread

File: src/screens/notifications/notifications.tsx (hunk 6 above)
Comment (lendihop) [reply src/screens/notifications/notifications.tsx:16]: look at the 1 comment thread

File: src/components/activity-groups-list/activity-groups-list.tsx (hunk 1 above)
Comment (lendihop) [reply src/components/activity-groups-list/activity-groups-list.tsx:24]: also flashList itself gives us this value in the console, if `estimatedItemSize` prop in not set

File: src/screens/collectibles-home/collectibles-list/collectibles-list.tsx [hunk 7]
```diff
@@ -20,39 +21,40 @@ const ITEMS_PER_ROW = 3;
...
 
-  const getItemLayout = createGetItemLayout<TokenInterface[]>(itemSize);
-
   const renderItem: ListRenderItem<TokenInterface[]> = useCallback(
     ({ item }) => (
       <View style={CollectiblesListStyles.rowContainer}>
-        {item.map(collectible => (
-          <TouchableCollectibleIcon key={getTokenSlug(collectible)} collectible={collectible} size={itemSize} />
+        {item.map((collectible, index) => (
+          <TouchableCollectibleIcon key={index} collectible={collectible} size={itemSize} />
```
[the remark this answers is not in the document]
Comment (lendihop) [reply src/screens/collectibles-home/collectibles-list/collectibles-list.tsx:41]: https://shopify.github.io/flash-list/docs/fundamentals/performant-components#remove-key-prop

File: android/build.gradle [hunk 8]
```diff
@@ -10,6 +10,8 @@ buildscript {
         Themis_compileSdkVersion = 33
         Themis_targetSdkVersion = 33
 
+        kotlinVersion = "1.8.20"
```
[the remark this answers is not in the document]
Comment (lendihop) [reply android/build.gradle:13]: no reason, set to latest

**Review by keshan3262 — DISMISSED** [review]

File: src/screens/collectibles-home/collectibles-list/collectibles-list.tsx [hunk 9]
```diff
@@ -20,39 +21,40 @@ const ITEMS_PER_ROW = 3;
...
     [itemSize]
   );
 
+  const ListEmptyComponent = useMemo(() => <DataPlaceholder text="Not found any NFT" />, []);
+
   return (
-    <FlatList
+    <FlashList
       data={data}
       renderItem={renderItem}
```
[the remark this answers is not in the document]
Comment (lendihop) [reply src/screens/collectibles-home/collectibles-list/collectibles-list.tsx:53]: `numColumns` sets its own gap between columns and there is no way to change it cause FlashList don't have `columnWrapperStyle` prop

File: src/screens/collectibles-home/collectibles-list/collectibles-list.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (lendihop) [reply src/screens/collectibles-home/collectibles-list/collectibles-list.tsx:41]: i agree)

**Review by keshan3262 — DISMISSED** [review]

File: src/components/collectible-icon/collectible-icon.tsx [hunk 10]
```diff
@@ -83,20 +92,36 @@ export const CollectibleIcon: FC<CollectibleIconProps> = ({
...
+  );
+
+  const lastItemId = useRef(assetSlug);
+
+  const [isLoadingFailed, setIsLoadingFailed] = useState<InitialLoadingState>(actualInitialLoadingState);
+  if (assetSlug !== lastItemId.current) {
+    lastItemId.current = assetSlug;
+    setIsLoadingFailed(actualInitialLoadingState);
+  }
+
```
Comment (lendihop) [inline src/components/collectible-icon/collectible-icon.tsx:116]: https://shopify.github.io/flash-list/docs/recycling

**Review by IrynaKhyzhynska — APPROVED** [review]
android done 1.16_999

**Review by NovakVlad2424 — APPROVED** [review]
ios done: 1.16_999

**Pull request merged**